### PR TITLE
initial commit for using dtl logger for dotnetcore build

### PR DESCRIPTION
**Pull request by tejasd1990**
#7749

### CODE REVIEW

File: Tasks/DotNetCoreCLIV2/task.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "version": {
         "Major": 2,
         "Minor": 162,
```
Comment (hiyadav) [inline Tasks/DotNetCoreCLIV2/task.json]: this should be the current sprint version

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts [hunk 2]
```diff
@@ -105,21 +108,43 @@ export class dotNetExe {
             } else {
                 dotnet.arg(projectFile);
             }
+            if(this.isBuildCommand()) {
```
Comment (hiyadav) [inline Tasks/DotNetCoreCLIV2/dotnetcore.ts]: Can this logger be used in other commands such as: publish/pack. As these commands inherently runs dotnet build

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts [hunk 3]
```diff
@@ -105,21 +108,43 @@ export class dotNetExe {
...
+                var detailId = "";
+                var projectFilePath = path.dirname(projectFile);
+                if(this.logProjectEvents) {
+                    detailId = uuidV4();
+                    var projectFileName = path.basename(projectFile);
+                    var detailName = tl.loc("Build", projectFileName);
+                    var startTime = new Date(Date.now()).toISOString();
+                    //Todo: can we use named parameters in typescipt?
+                    tl.logDetail(detailId, "", "", "Process", detailName, 0, startTime, "", 0, tl.TaskState.Initialized);
```
Comment (hiyadav) [inline Tasks/DotNetCoreCLIV2/dotnetcore.ts]: what would this do

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts [hunk 4]
```diff
@@ -105,21 +108,43 @@ export class dotNetExe {
...
+                }
+                var loggerAssembly = path.join(__dirname, 'dotnet-build-helpers/Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll');
```
Comment (hiyadav) [inline Tasks/DotNetCoreCLIV2/dotnetcore.ts]: Does this imply that the new logger will now be used for each build ?

Comment (tejasd1990) [reply Tasks/DotNetCoreCLIV2/dotnetcore.ts]: Yes. There was no logger previously, so we used to get the command output as it is.

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts (hunk 3 above)
Comment (tejasd1990) [reply Tasks/DotNetCoreCLIV2/dotnetcore.ts]: This creates a [timeline record](https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#logdetail-create-and-update-a-timeline-record-for-a-task) for the command, which is a recursive log according to the operations the command executes. This particular logger just prints the method names run as part of the command. It helps in debugging. This behaviour is present in msbuild and vsbuild tasks as well

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts (hunk 2 above)
Comment (tejasd1990) [reply Tasks/DotNetCoreCLIV2/dotnetcore.ts]: Yes it should be, then we will get warnings highlighted for those commands as well.

**Comment by bishal-pdMSFT** [comment]
@tejasd1990 should this be abandoned?

**Pull request closed without merging**